package/src/main.js

```diff
@@ -42,8 +40,27 @@ Drupal.behaviors.audiofile.attach = function (context, settings) {
           $fid.val(data['fid'])
           $signature.val(data['signature'])
           $url.val(data['url'])
+        },
+        complete: (data, status) => {
+          requestTracker.done()
         }
-      })
-    }).appendTo(element)
+      }))
+    })
+
+    $(element).on('audiofile:reset', (event) => {
+      requestTracker.abort()
+      $fid.val('')
+      $signature.val('')
+      $url.val('')
+    })
+
+    // Disable submit button during file upload.
+    $(element).on('request:start', (event, data) => {
+      $('[type="submit"]', form).prop('disabled', true)
+    })
+    $(element).on('request:end', (event, data) => {
+      $('[type="submit"]', form).prop('disabled', false)
+    })
```

Comment:
This is assuming that no submit-buttons are disabled for other reasons, right?

Reply:
I'm afraid so, yes.